substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/c/CAnnotationProcessor.java

```diff
@@ -71,6 +73,12 @@ public CAnnotationProcessor(NativeLibraries nativeLibs, NativeCodeContext codeCt
         }
         this.compilerInvoker = ImageSingletons.lookup(CCompilerInvoker.class);
         this.tempDirectory = compilerInvoker.tempDirectory;
+        this.queryCodeDirectory = compilerInvoker.tempDirectory;
+
+        String queryCodeDir = CAnnotationProcessorCache.Options.CAPQueryCodeDir.getValue();
+        if (!queryCodeDir.equals("")) {
```

Comment:
can queryCodeDir be `null` ? 
Better to use isEmpty()

Reply:
Good idea, changed!